src/libstd/io/mod.rs

```diff
@@ -558,6 +558,12 @@ pub trait BufRead: Read {
     /// This function does not perform any I/O, it simply informs this object
     /// that some amount of its buffer, returned from `fill_buf`, has been
     /// consumed and should no longer be returned.
+    ///
+    /// This function is used to tell the buffer how many bytes you've consumed from the return
+    /// value of `fill_buf`, and so may do odd things if `fill_buf` isn't called before calling
+    /// this.
```

Comment:
Can this follow the surrounding style of wrapping to 80 characters? Other than that r=me